docs/tutorials/PyTorch.py

```diff
@@ -0,0 +1,322 @@
+        -0.8388608,
+        -0.67108864,
+        -0.536870912,
+        -0.42949673,
+        -0.343597384,
+        -0.274877907,
+        -0.219902326,
+        -0.17592186,
+        -0.140737488,
+        -0.112589991,
+        -0.090071993,
+        -0.072057594,
+        -0.057646075,
+        -0.04611686,
+    ]
+)
+y_vals = np.array(
+    [
+        16,
+        10.24,
+        6.5536,
+        4.194304,
+        2.68435456,
+        1.717986918,
+        1.099511628,
+        0.703687442,
+        0.450359963,
+        0.288230376,
+        0.184467441,
+        0.118059162,
+        0.075557864,
+        0.048357033,
+        0.030948501,
+        0.019807041,
+        0.012676506,
+        0.008112964,
+        0.005192297,
+        0.00332307,
+        0.002126765,
+    ]
+)
+
+plt.scatter(x_vals, y_vals)
+
+plt.xlim(xmin=-5, xmax=5)
+plt.ylim(ymin=0, ymax=25)
+plt.show()
+# -
+
+# This works, but it takes a long time since we have several small steps. We could speed up the process by taking large steps. Starting at (-4,16) and increasing the step of size to 1.5 we find:
+
+# +
+q = np.linspace(-20, 20, 1000)
+plt.plot(q, q ** 2)
+
+
+x_vals = np.array([-4, 8])
+y_vals = np.array([16, 64])
+
+plt.plot(x_vals, y_vals)
+
+
+plt.xlim(xmin=-20, xmax=20)
+plt.ylim(ymin=0, ymax=260)
+plt.show()
+# -
+
+# *Note the change in scale.* With only one step, we already see that we stepped over the minimum! If we look at the tangent line at this point we see that the direction of the steepest descent is now to the left instead of the right:
+
+# +
+q = np.linspace(-20, 20, 1000)
+plt.plot(q, q ** 2)
+
+
+plt.scatter(-4, 16)  # point
+qp = -4
+yp = 16
+
+
+plt.scatter(8, 64)  # second point
+qp2 = 8
+yp2 = 64
+
+
+plt.plot(q, qp2 * 2 * (q - qp2) + yp2)  # tangent line
+
+plt.xlim(xmin=-20, xmax=20)
+plt.ylim(ymin=0, ymax=260)
+plt.show()
+# -
+
+# In our attempt to continue to find the minimum, we would now take a step of the same size, but now to the left. This puts us back at on the left side of the hill, but now above where we started. We would be stuck going back and forth between the two sides of the hill continuing to go upward.
+
+# We know as we get closer to the minimum the slope/gradient levels out. This means if we are very far from the minimum, we know we can take a very large step without stepping over the minimum. However, if we are close to the minimum, with a very small slope, we need to be careful and take smaller steps.
+#
+# We would continue this process of (1) Calculating our slope, (2) Determining a step size, and (3) Moving that step size until we finally reached a spot where the slope or gradient is essentially zero.
+#
+# This process is a gradient descent and it is used to minimize functions.
+
+# +
+q = np.linspace(-20, 20, 1000)
+plt.plot(q, q ** 2)
+
+
+x_vals = np.array([-4, 8, -16])
+y_vals = np.array([16, 64, 256])
+
+plt.plot(x_vals, y_vals)
+
+
+plt.xlim(xmin=-20, xmax=20)
+plt.ylim(ymin=0, ymax=260)
+plt.show()
```

Comment:
Since you went to the trouble of saving qp, yp, qp2, yp2, etc., I think you should use them in `x_vals` and `y_vals` instead of hard-coding the values. Otherwise it's more difficult to see exactly where these numbers are coming from.

I also think this plot needs more description, since I don't see an explanation of what `plt.plot(x_vals, y_vals)` is plotting. I think it's pretty clear what you're plotting here, but adding a sentence explicitly stating that is probably good practice.